File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int m = bank[0].length();
        int tmp;
        int pre = 0;
        int res = 0;
        for (String s : bank) {
            tmp = 0;
            for (int j = 0; j < m; j++) {
                if (s.charAt(j) == '1') {
                    tmp++;
                }
            }
            if (tmp > 0) {
                res += tmp * pre;
                pre = tmp;
            }
        }
        return res;
    }
}

File: 2125-number-of-laser-beams-in-a-bank/2125-number-of-laser-beams-in-a-bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        n = len(bank)
        m = len(bank[0])
        pre = 0
        res = 0
        for i in range(n):
            tmp = 0
            for j in range(m):
                if bank[i][j] == "1":
                    tmp += 1
            if tmp:
                res += tmp*pre
                pre = tmp
            
        return res